frontend/src/component/Activity/Preview.jsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import { Link } from 'react-router-dom'
+import {
+  buildFilePreviewUrl as jpgPreviewUrl,
+  removeExtensionOfFilename,
+  removeInteractiveContentFromHTML,
+  SCREEN_SIZE
+} from 'tracim_frontend_lib'
+import { FETCH_CONFIG, PAGE } from '../../util/helper.js'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+
+require('./Preview.styl')
+
+// possible preview sizes with their media queries, ordered from smallest to biggest
+const PREVIEW_WIDTHS = [
+  [
+    `(max-width: ${SCREEN_SIZE.MAX_SM})`,
+    376 // 400px - margins (10 * 2px) - borders (2 * 1px)
+  ],
+  [
+    '',
+    776 // 800px - margins (10 * 2px) - borders (2 * 1px)
+  ]
+]
+
+const MAX_PREVIEW_HEIGHT = 300
+
+class Preview extends React.Component {
+  state = {
+    preview: this.isHtmlPreview() ? null : this.getJPEGPreview(),
+    previewOverflow: false,
+    previewLoading: this.isHtmlPreview(),
+    previewUnavailable: false
+  }
+
+  isHtmlPreview () {
+    const type = this.props.content.content_type
+    return type === 'html-document' || type === 'thread'
+  }
+
+  getHTMLPreviewUrl ({ content_type: type, content_id: id, workspace_id: spaceId, label }) {
+    return `${FETCH_CONFIG.apiUrl}/workspaces/${spaceId}/${type}s/${id}/preview/html/${label}.html`
+  }
+
+  handleUnavailablePreview = () => {
+    this.setState({
+      previewLoading: false,
+      preview: null,
+      previewUnavailable: true
+    })
+  }
+
+  setLoadingPreview () {
+    this.setState({ previewLoading: true })
+  }
+
+  async getHTMLPreview () {
+    const fetchResult = await fetch(this.getHTMLPreviewUrl(this.props.content), {
```

Comment:
Suggestion, if you apply, do not forget to propagate to the rest of the function.
```suggestion
    const fetchResultGetHTMLPreviewUrl = await fetch(this.getHTMLPreviewUrl(this.props.content), {
```